src/mjolnir/landmark_builder.cc

```diff
@@ -220,6 +228,80 @@ void LandmarkDatabase::release_prepared_stmt() {
   LOG_INFO("released prepared statements");
 }
 
+// Task 2: parse landmarks
+// anonymous namespace?
+struct landmark_callback : public OSMPBF::Callback {
+public:
+  landmark_callback(LandmarkDatabase& db) : db_(db) {
+  }
+  virtual ~landmark_callback() {
+  }
+
+  virtual void
+  node_callback(const uint64_t /*osmid*/, double lng, double lat, const OSMPBF::Tags& tags) override {
+    Landmark landmark;
+
+    for (const auto& tag : tags) {
+      if (tag.first == "amenity") {
+        // if amenity is empty will return LandmarkType::null
+        landmark.type = string_to_landmark_type(tag.second);
+      }
+      if (tag.first == "name" && !tag.second.empty()) {
+        landmark.name = tag.second;
+      }
+      landmark.lng = lng;
+      landmark.lat = lat;
+    }
+    // insert parsed landmark directly into database
+    db_.insert_landmark(landmark);
```

Comment:
doesn't this save _all_ nodes right now?! 

Also I'd not handle the case where `amenity` has no value. I'm not even sure that can happen in actual OSM data (but surely can in custom OSM data). Even there is smth like that, it has no value for us IMO.

Reply:
We could instead do a LOG_ERROR when there's an amenity node without a value.

Reply:
agreed, it has to have a type otherwise its meaningless

Reply:
thanks for pointing this out! I totally forgot to include landmark nodes only and made a premise that not all landmark nodes have amenity value which is incorrect. will fix that.

Reply:
a further point: in the landamark database should we only store nodes that belong to `enum class LandmarkType`? Now the list contians 21 popular landmark types and others will be translated into `LandmarkType::null` which is useless in navigation. plus we can shorten execution time and save storage by storing "known" nodes only.

Reply:
exactly right, we should skip anything that is useless, no point in even looking at it if we cant parse the string into one of the enums that we accept

Reply:
got it. done.